File: summer/WebContent/js/my/ColumnDefinition.js
/**
 * ColumnDefinition
 */

define(["dojo/_base/declare", "system/Type", "controls/DefinitionBase"], 
		function(declare, Type, DefinitionBase){
	var ColumnDefinition = declare("ColumnDefinition", DefinitionBase,{
		constructor:function(){
		}
	});
	
	Object.defineProperties(ColumnDefinition.prototype,{
		  
	});
	
	Object.defineProperties(ColumnDefinition,{
		  
	});
	
	ColumnDefinition.Type = new Type("ColumnDefinition", ColumnDefinition, [DefinitionBase.Type]);
	return ColumnDefinition;
});




using MS.Internal.PresentationFramework;
using System;
using System.ComponentModel;
namespace System.Windows.Controls
{
	public class ColumnDefinition : DefinitionBase
	{
		[CommonDependencyProperty]
		public static readonly DependencyProperty WidthProperty = DependencyProperty.Register("Width", typeof(GridLength), typeof(ColumnDefinition), new FrameworkPropertyMetadata(new GridLength(1.0, GridUnitType.Star), new PropertyChangedCallback(DefinitionBase.OnUserSizePropertyChanged)), new ValidateValueCallback(DefinitionBase.IsUserSizePropertyValueValid));
		[CommonDependencyProperty, TypeConverter("System.Windows.LengthConverter, PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, Custom=null")]
		public static readonly DependencyProperty MinWidthProperty = DependencyProperty.Register("MinWidth", typeof(double), typeof(ColumnDefinition), new FrameworkPropertyMetadata(0.0, new PropertyChangedCallback(DefinitionBase.OnUserMinSizePropertyChanged)), new ValidateValueCallback(DefinitionBase.IsUserMinSizePropertyValueValid));
		[CommonDependencyProperty, TypeConverter("System.Windows.LengthConverter, PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, Custom=null")]
		public static readonly DependencyProperty MaxWidthProperty = DependencyProperty.Register("MaxWidth", typeof(double), typeof(ColumnDefinition), new FrameworkPropertyMetadata(double.PositiveInfinity, new PropertyChangedCallback(DefinitionBase.OnUserMaxSizePropertyChanged)), new ValidateValueCallback(DefinitionBase.IsUserMaxSizePropertyValueValid));
		public GridLength Width
		{
			get
			{
				return base.UserSizeValueCache;
			}
			set
			{
				base.SetValue(ColumnDefinition.WidthProperty, value);
			}
		}
		[TypeConverter(typeof(LengthConverter))]
		public double MinWidth
		{
			get
			{
				return base.UserMinSizeValueCache;
			}
			set
			{
				base.SetValue(ColumnDefinition.MinWidthProperty, value);
			}
		}
		[TypeConverter(typeof(LengthConverter))]
		public double MaxWidth
		{
			get
			{
				return base.UserMaxSizeValueCache;
			}
			set
			{
				base.SetValue(ColumnDefinition.MaxWidthProperty, value);
			}
		}
		public double ActualWidth
		{
			get
			{
				double result = 0.0;
				if (base.InParentLogicalTree)
				{
					result = ((Grid)base.Parent).GetFinalColumnDefinitionWidth(base.Index);
				}
				return result;
			}
		}
		public double Offset
		{
			get
			{
				double result = 0.0;
				if (base.Index != 0)
				{
					result = base.FinalOffset;
				}
				return result;
			}
		}
		public ColumnDefinition() : base(true)
		{
		}
	}
}

File: org.summer.view.widget/src/org/summer/view/widget/controls/ColumnDefinition.java
/**
 */
package org.summer.view.widget.controls;

import org.summer.view.widget.DependencyObject;
import org.summer.view.widget.DependencyProperty;
import org.summer.view.widget.DependencyPropertyChangedEventArgs;
import org.summer.view.widget.FrameworkPropertyMetadata;

public class ColumnDefinition extends DefinitionBase {
//    #region Public Fields
//    #region Dependency Properties
//    [TypeConverter("System.Windows.LengthConverter, PresentationFramework, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, Custom=null")]
    public static final DependencyProperty MaxWidthProperty = DependencyProperty.Register("MaxWidth", typeof(double), typeof(ColumnDefinition), new FrameworkPropertyMetadata(double.PositiveInfinity, InvalidateGridMeasure));
//    [TypeConverter("System.Windows.LengthConverter, PresentationFramework, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, Custom=null")]
    public static final DependencyProperty MinWidthProperty = DependencyProperty.Register("MinWidth", typeof(double), typeof(ColumnDefinition), new FrameworkPropertyMetadata(InvalidateGridMeasure));
    public static final DependencyProperty WidthProperty = DependencyProperty.Register("Width", typeof(GridLength), typeof(ColumnDefinition), new FrameworkPropertyMetadata(new GridLength(1, GridUnitType.Star), InvalidateGridMeasure));
//    #endregion
//    #endregion

//    #region Private Fields
    double actual_width;
    double offset;
    Grid grid;
//    #endregion

//    #region Public Constructors
    public ColumnDefinition() {
    }
//    #endregion

//    #region Public Properties
//    #region Dependency Properties
//    [TypeConverter(typeof(LengthConverter))]
    public double MaxWidth {
            get { return (double)GetValue(MaxWidthProperty); }
            set { SetValue(MaxWidthProperty, value); } 
    }

//    [TypeConverter(typeof(LengthConverter))]
    public double MinWidth {
            get { return (double)GetValue(MinWidthProperty); }
            set { SetValue(MinWidthProperty, value); }
    }

    public GridLength Width {
            get { return (GridLength)GetValue(WidthProperty); }
            set { SetValue(WidthProperty, value); }
    }
//    #endregion

    public double ActualWidth {
            get { return actual_width; }
            /*internal*/ public set { actual_width = value; }
    }

    public double Offset {
            get { return offset; }
            /*internal*/ public set { offset = value; }
    }
//    #endregion

//    #region Internal Properties
    /*internal*/ public Grid Grid {
            get { return grid; }
            set { grid = value; }
    }
//    #endregion

//    #region Private Methods
    static void InvalidateGridMeasure(DependencyObject d, DependencyPropertyChangedEventArgs e) {
            Grid grid = ((ColumnDefinition)d).grid;
            if (grid != null)
                    grid.InvalidateMeasure();
    }
//    #endregion
}